### DISCUSSION TITLE
Are realms the only way to round trip a mutable object from Rust -> JS and back?

### DISCUSSION THREAD

**Discussion by tom-sherman** [boa-dev/boa#3819] [Q&A]
I want to pass an object into the default function export of a `Module`. That function will mutate the object I pass in. When that function has finished executing (or the promise it returns has resolved) I want to use that updated object on the Rust side.

I saw the example using realms but was wondering if there's an easier way.

**Comment by jedel1043** [comment]
For now yes, but I think with https://github.com/boa-dev/boa/pull/3802 you will be able to inject data to the context itself.

**Reply by tom-sherman** [reply]
Thanks. Are there any examples of what I'm trying to achieve here?

I wanna insert an object in, allow JS to modify it, and then read it back out. I'm happy to clone/snapshot the object for that last step.
